server/src/analyzer/typeUtils.ts

```diff
@@ -1143,4 +1143,13 @@ export class TypeUtils {
 
         return unionType;
     }
+
+    static isDataClass(type: ClassType): boolean {
+      return (type.getBaseClasses().length === 1 && type.getBaseClasses()[0].type.asString() === 'Type[NamedTuple]');
```

Comment:
This looks like a very fragile approach. What if NamedTuple isn't an immediate base class? Also, we shouldn't use the string representation (which is meant only as a human-readable form) for internal logic.

Reply:
I agree, but am out of ideas.  Preferably a python like 'isinstance()' type of check needs to be performed here.  If there are any internal apis that perform this, please point them out.

Reply:
On further check, Python 3.7 accepts this syntax if only if NamedTuple is one of the direct base classes, there is no need to scan ancestor tree. The code should be:
```ts
    static isDataClass(type: ClassType): boolean {
        return type.getBaseClasses().some( v => v.type.asString() === 'Type[NamedTuple]');
    }
```
Now regarding the string literal check "v.type.asString() === 'Type[NamedTuple]'",  if there is a better check, please let me know.

Reply:
The check is changed to:
```ts
        return (type.getBaseClasses().some( v => v.type instanceof ClassType && v.type.isBuiltIn()
            && v.type.getClassName() === 'NamedTuple' ));
```